superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx

```diff
@@ -100,29 +99,36 @@ export default function PluginFilterSelect(props: PluginFilterSelectProps) {
     defaultToFirstItem,
     searchAllOptions,
   } = formData;
+  const groupby = ensureIsArray<string>(formData.groupby);
+  const [col] = groupby;
+  const [currentSuggestionSearch, setCurrentSuggestionSearch] = useState('');
+  const [dataMask, dispatchDataMask] = useReducer<DataMaskReducer>(reducer, {
+    filterState,
+    ownState: {
+      coltypeMap,
+    },
+  });
+  const updateDataMask = (values: SelectValue) => {
+    const emptyFilter =
+      enableEmptyFilter && !inverseSelection && values?.length === 0;
+
+    dispatchDataMask({
+      type: 'filterState',
+      extraFormData: getSelectExtraFormData(
+        col,
+        values,
+        emptyFilter,
+        inverseSelection,
+      ),
+      filterState: {
+        value: values,
+      },
+    });
+  };
 
   const isDisabled =
     appSection === AppSection.FILTER_CONFIG_MODAL && defaultToFirstItem;
 
-  const groupby = ensureIsArray<string>(formData.groupby);
-  // Correct initial value for Ant Select
-
-  // If we are in config modal we always need show empty select for `defaultToFirstItem`
-  const [values, setValues] = useState<SelectValue>(
-    !isDisabled && defaultValue?.length ? defaultValue : [],
-  );
```

Comment:
state handling is now deferred to the application, i.e. filter state is only being set via `setDataMask`.